src/server/utils/user.js

```diff
@@ -0,0 +1,68 @@
+/* Return a new user object based on fields.
+ *
+ * This assumes that the field set was already mapped through mapColumns(...,
+ * 'auth'). It is mostly of use to the login manager; afterward, the object
+ * should be stored as req.session.user.
+ *
+ * As this is stored in the session, ensure that keys and values stored here
+ * are serializeable. In particular, ensure this object can losslessly be
+ * passed via JSON.
+ */
+function newUser(fields) {
+  // Shallow copy the user object to avoid reference woes.
+  // NB: Don't put any objects in here that you don't want sharing references.
+  // Arrays are special-cased.
+  var user = {};
+  Object.keys(fields).map(key => {
+    if(Array.isArray(fields[key])) {
+      user[key] = fields[key].slice();
+    } else {
+      user[key] = fields[key];
+    }
+  });
+
+  // Set defaults.
+  if(user.permissions === undefined) user.permissions = [];
+
+  // Create a permission set for faster lookups.
+  user.perms = {};
+  for(var i = 0; i < user.permissions.length; i++) {
+    user.perms[user.permissions[i]] = true;
+  }
+
+  // Delete these keys both to avoid unnecessary traffic and to not reveal
+  // useless information to the client.
+  delete user.vendors;
+  delete user.permissions;
+
+  return user;
+}
+
+const PERM = {
```

Comment:
Go ahead and use the full word or phrase here: `PERMISSIONS` or `PERMISSION_TYPE`